corehq/apps/reports/views.py

```diff
@@ -162,6 +164,24 @@
 require_can_view_all_reports = require_permission(HqPermissions.view_reports)
 
 
+def _augmented_form_view_permissions(login_decorator=login_and_domain_required):
+    def decorator(view_func):
+        @wraps(view_func)
+        def _inner(request, domain, *args, **kwargs):
+            if VIEW_FORM_ATTACHMENT.enabled(domain):
+                return view_func(request, domain, *args, **kwargs)
+            return require_form_view_permission(view_func)(request, domain, *args, **kwargs)
+
+        if login_decorator:
+            return login_decorator(_inner)
+        else:
+            return _inner
```

Comment:
What's the use-case here?  When might you not want these views protected by login?  Also, I think the login decorator can be applied just in the `VIEW_FORM_ATTACHMENT.enabled` branch, since `require_form_view_permission` already checks domain access.

Reply:
I've done that because the `require_form_view_permission` login_decorator is set to `None` [here](https://github.com/dimagi/commcare-hq/blob/cabd9fe44b06ad753f160f763c4704f89a76a5f9/corehq/apps/reports/views.py#L161), unless I'm missing something?

Reply:
Ah, okay this makes a bit more sense now - I wasn't aware of that pattern of setting `login_decorator` to `None`, so I see now that you're following that.  I'm not sure why we do that, and I'm a little suspicious of it.  At any rate, in those instances, the permissions check is still applied, and that check will also fail if the user is not a member of the domain:
https://github.com/dimagi/commcare-hq/blob/fb31bc8b02c29b26bdd1c69ea1c916533ac4b119/corehq/apps/users/models.py#L537-L540
If you look at usages of `@require.*permission`, some use a separate login decorator as well, and some don't - seems to be a pretty even split.  That this is acceptable seems to me like almost an accident, actually, and it's not how I'd design the system from scratch.

That said, in the current state, it is not required to use a login decorator when using a `require_permissions` decorator, so I think it makes sense to stick to that pattern until we fix it.  This new decorator circumvents the permissions check when the feature flag is enabled, meaning it _also_ circumvents any domain checks, which violates the pattern seen elsewhere.

Reply:
@esoergel Ah, good point. I completely skipped that part. I made an [update](https://github.com/dimagi/commcare-hq/pull/31936/commits/44607b5dc17cfa6b7dcd9b5c48ecb0fc0e122cf6) which also checks for user domain membership when the FF is enabled for that domain.